contracts/zero-ex/contracts/test/foundry/utils/ForkUtils.sol

```diff
@@ -0,0 +1,343 @@
+        for (uint256 i = 0; i < chains.length; i++) {
+            forkIds[chains[i]] = vm.createFork(vm.rpcUrl(chains[i]), blockNumber[i]);
+        }
+    }
+
+    //gets a dummy signer to be used for an OTC order
+    function getSigner() public returns (address, uint) {
+        string memory mnemonic = "test test test test test test test test test test test junk";
+        uint256 privateKey = vm.deriveKey(mnemonic, 0);
+
+        vm.label(vm.addr(privateKey), "zeroEx/MarketMaker");
+        return (vm.addr(privateKey), privateKey);
+    }
+
+    //read the uniswapV2 router addresses from file
+    function readLiquiditySourceAddresses() internal returns (string memory) {
+        string memory root = vm.projectRoot();
+        string memory path = string(
+            abi.encodePacked(root, "/", "contracts/test/foundry/addresses/SourceAddresses.json")
+        );
+        sourcesJson = vm.readFile(path);
+        return vm.readFile(path);
+    }
+
+    //retrieve the uniswapV2 router addresses
+    function getLiquiditySourceAddresses(uint index) public returns (LiquiditySources memory sources) {
+        readLiquiditySourceAddresses();
+        bytes memory liquiditySources = sourcesJson.parseRaw(indexChainIds[index]);
+        return abi.decode(liquiditySources, (LiquiditySources));
+    }
+
+    function readAddresses() internal returns (string memory) {
+        string memory root = vm.projectRoot();
+        string memory path = string(abi.encodePacked(root, "/", "contracts/test/foundry/addresses/addresses.json"));
+        addressesJson = vm.readFile(path);
+        return vm.readFile(path);
+    }
+
+    //retrieve the 0xProtocol contract addresses from addresses.json
+    function getContractAddresses(uint index) public returns (Addresses memory addresses) {
+        readAddresses();
+        bytes memory contractAddresses = addressesJson.parseRaw(indexChainIds[index]);
+        return abi.decode(contractAddresses, (Addresses));
+    }
+
+    function readTokens() internal returns (string memory) {
+        string memory root = vm.projectRoot();
+        string memory path = string(
+            abi.encodePacked(root, "/", "contracts/test/foundry/addresses/TokenAddresses.json")
+        );
+        tokensJson = vm.readFile(path);
+        return vm.readFile(path);
+    }
+
+    //retrieve our tokenList from TokenAddresses.json
+    function getTokens(uint index) public returns (TokenAddresses memory addresses) {
+        readTokens();
+        bytes memory chainTokens = tokensJson.parseRaw(indexChainIds[index]);
+        return abi.decode(chainTokens, (TokenAddresses));
+    }
+
+    //creates the appropriate bridge adapter based on what chain the tests are currently executing on.
+    function createBridgeAdapter(IEtherTokenV06 weth) public returns (IBridgeAdapter bridgeAdapter) {
+        uint chainId;
+
+        assembly {
+            chainId := chainid()
+        }
+        if (chainId == uint256(address(0))) {
```

Comment:
I'm not sure what this is intended to do? For example, if you ran these tests on `Goerli` you'd get a non-0 `chainId`, it would pass this check, but then create a `BridgeAdapter(address(0))`

The check should be something like

```solidity
        } else if (chainId == 42161) {
            return IBridgeAdapter(new ArbitrumBridgeAdapter(weth));
        } else {
            revert("CHAIN ID NOT MAPPED");
        }
```